File: Problems & Solutions/144. Binary Tree Preorder Traversal/09 January 2022/Approach 2 - Iteration/Binary Tree Preorder Traversal.py
class Solution:
    def preorderTraversal(self, root: TreeNode) -> List[int]:
        answer = []
        stack = [root]

        while stack:
            curr_node = stack.pop()
            if curr_node:
                answer.append(curr_node.val)
                stack.append(curr_node.right)
                stack.append(curr_node.left)
                
        return answer

File: Problems & Solutions/144. Binary Tree Preorder Traversal/09 January 2022/Approach 2 - Iteration/Binary Tree Preorder Traversal.java
class Solution {    
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> answer = new ArrayList<>();
        Stack<TreeNode> stack = new Stack<>();
        stack.add(root);
        
        while (!stack.isEmpty()) {
            TreeNode currNode = stack.peek();
            stack.pop();
            if (currNode != null) {
                answer.add(currNode.val);
                stack.add(currNode.right);
                stack.add(currNode.left);
            }
        }
        
        return answer;
    }
}